map/bookmark.cpp

```diff
@@ -182,418 +156,396 @@ size_t BookmarkCategory::FindBookmark(Bookmark const * bookmark) const
-      while (cortegeIter)
-      {
-        m2::PointD pt;
-        if (ParsePoint(*cortegeIter, coordSeparator, pt))
-          m_points.Add(pt);
-        ++cortegeIter;
-      }
+  void ParseLineCoordinates(string const & s, char const * blockSeparator,
+                            char const * coordSeparator)
+  {
+    m_geometryType = LINE;
+
+    strings::SimpleTokenizer cortegeIter(s, blockSeparator);
+    while (cortegeIter)
+    {
+      m2::PointD pt;
+      if (ParsePoint(*cortegeIter, coordSeparator, pt))
+        m_points.Add(pt);
+      ++cortegeIter;
     }
+  }
 
-    bool MakeValid()
-    { 
-      if (POINT == m_geometryType)
+  bool MakeValid()
+  {
+    if (POINT == m_geometryType)
+    {
+      if (MercatorBounds::ValidX(m_org.x) && MercatorBounds::ValidY(m_org.y))
       {
-        if (MercatorBounds::ValidX(m_org.x) && MercatorBounds::ValidY(m_org.y))
-        {
-          // set default name
-          if (m_name.empty())
-            m_name = PointToString(m_org);
+        // set default name
+        if (m_name.empty())
+          m_name = PointToString(m_org);
 
-          // set default pin
-          if (m_type.empty())
-            m_type = "placemark-red";
+        // set default pin
+        if (m_type.empty())
+          m_type = "placemark-red";
 
-          return true;
-        }
-        return false;
-      }
-      else if (LINE == m_geometryType)
-      {
-        return m_points.GetSize() > 1;
+        return true;
       }
-
       return false;
     }
-
-    void ParseColor(string const & value)
+    else if (LINE == m_geometryType)
     {
-      string fromHex = FromHex(value);
-      ASSERT(fromHex.size() == 4, ("Invalid color passed"));
-      // Color positions in HEX – aabbggrr
-      m_trackColor = graphics::Color(fromHex[3], fromHex[2], fromHex[1], fromHex[0]);
+      return m_points.GetSize() > 1;
     }
 
-    bool GetColorForStyle(string const & styleUrl, graphics::Color & color)
-    {
-      if (styleUrl.empty())
-        return false;
+    return false;
+  }
 
-      // Remove leading '#' symbol
-      map<string, graphics::Color>::const_iterator it = m_styleUrl2Color.find(styleUrl.substr(1));
-      if (it != m_styleUrl2Color.end())
-      {
-        color = it->second;
-        return true;
-      }
-      return false;
-    }
+  void ParseColor(string const & value)
+  {
+    string fromHex = FromHex(value);
+    ASSERT(fromHex.size() == 4, ("Invalid color passed"));
+    // Color positions in HEX – aabbggrr
+    m_trackColor = graphics::Color(fromHex[3], fromHex[2], fromHex[1], fromHex[0]);
+  }
 
-  public:
-    KMLParser(BookmarkCategory & cat) : m_category(cat)
-    {
-      Reset();
-    }
+  bool GetColorForStyle(string const & styleUrl, graphics::Color & color)
+  {
+    if (styleUrl.empty())
+      return false;
 
-    bool Push(string const & name)
+    // Remove leading '#' symbol
+    map<string, graphics::Color>::const_iterator it = m_styleUrl2Color.find(styleUrl.substr(1));
```

Comment:
Please, add a dot at the end of the comment, and replace full iterator typename by auto.